### PR TITLE
xds: add support for setting bootstrap file with java system property

### PR DESCRIPTION
**Pull request by erikjoh**
While most languages support setting environment variables during runtime,
Java does not. In Java, the preferred approach is to use Java System Properties
in order so specify configuration options. By checking for the existence
of the io.grpc.xds.bootstrap property if GRPC_XDS_BOOTSTRAP is not found,
it is possible to either supply the bootstrap location during runtime or as
a Java argument. The environment variable still takes precedence in order
to not break any existing documentation.

For discussion see #7605 .

### CODE REVIEW

**Comment by erikjoh** [comment]
Blocked waiting for Corporate CLA approval.

**Comment by erikjoh** [comment]
I signed it

**Comment by erikjoh** [comment]
@ejona86 @sanjaypujare FYI the link here https://github.com/grpc/grpc-java/blob/master/CONTRIBUTING.md#legal-requirements (https://identity.linuxfoundation.org/projects/cncf) goes to the old and now outdated system.
We finally figured out how to get me signed though so now this PR is ready for review :slightly_smiling_face:

**Review by ejona86 — APPROVED** [review]
It seems like maybe we don't need to mention the system property in the example, but we don't actually have another place to put that documentation so it seems best for now.

File: xds/src/main/java/io/grpc/xds/Bootstrapper.java
Comment (sanjaypujare) [inline xds/src/main/java/io/grpc/xds/Bootstrapper.java:62]: Technically it's not necessary to remove this and use the `XdsLogger logger` variable. You can just replace the hardcoded string `BOOTSTRAP_PATH_SYS_ENV_VAR + "={0}"` with a String variable which is set to either `BOOTSTRAP_PATH_SYS_PROPERTY_VAR + "={0}"` or `BOOTSTRAP_PATH_SYS_ENV_VAR + "={0}"` in your 'if-else` statement. But this also works...

File: examples/example-xds/README.md
Comment (sanjaypujare) [inline examples/example-xds/README.md:27]: should we say something like below to be more precise?:

...`GRPC_XDS_BOOTSTRAP` environment variable (preferred) or if that is not set then the `io.grpc.xds.bootstrap` java system property...

**Review by sanjaypujare — APPROVED** [review]
Added 2 comments but looks good.

File: xds/src/main/java/io/grpc/xds/Bootstrapper.java
Comment (erikjoh) [reply xds/src/main/java/io/grpc/xds/Bootstrapper.java:62]: Hmm, I did it like this to avoid having a temporary variable only used for the log message but maybe that's fine in this case. What about something like this?
```
String filePathSource = BOOTSTRAP_PATH_SYS_ENV_VAR;
String filePath = System.getenv(filePathSource);
if(filePath == null) {
  filePathSource = BOOTSTRAP_PATH_SYS_PROPERTY_VAR;
  filePath = System.getProperty(filePathSource);
}
if(filePath == null) {
  throw new XdsInitializationException(
         "Environment variable " + BOOTSTRAP_PATH_SYS_ENV_VAR
         + " or Java System Property " + BOOTSTRAP_PATH_SYS_PROPERTY_VAR
         + " not defined.");
} 
XdsLogger
   .withPrefix(LOG_PREFIX)
   .log(XdsLogLevel.INFO, filePathSource + "={0}", filePath);
```

File: examples/example-xds/README.md
Comment (erikjoh) [reply examples/example-xds/README.md:27]: Makes sense!
I'll update it :slightly_smiling_face:

File: xds/src/main/java/io/grpc/xds/Bootstrapper.java
Comment (sanjaypujare) [reply xds/src/main/java/io/grpc/xds/Bootstrapper.java:62]: Sounds good. It's fine even if you don't change it since the logic is correct

**Comment by sanjaypujare** [comment]
@erikjoh it is better if you don't squash commits and "force-push" them - that way it is easier to see the changes made in the second commit. While merging the commits can be squashed.

**Review by sanjaypujare — APPROVED** [review]
LG

**Comment by erikjoh** [comment]
> @erikjoh it is better if you don't squash commits and "force-push" them - that way it is easier to see the changes made in the second commit. While merging the commits can be squashed.

Ah sorry, I don't use the squash merge feature a lot! Noted for the next time :slightly_smiling_face: 
You'll have to merge it once the build finishes since I don't have permission to do so.
Thanks for taking the time to have a look at this!

**Comment by erikjoh** [comment]
@sanjaypujare That build failure seems unrelated to this since the class was introduced recently and does not exist on my branch even. Doesn't seem like I have any option to rerun the build.

```
io.grpc.xds.SharedCallCounterMapTest > autoCleanUp FAILED
    expected to be empty
    but was: {cluster-foo.googleapis.com={null=io.grpc.xds.SharedCallCounterMap$CounterReference@21c8125f}}
        at io.grpc.xds.SharedCallCounterMapTest.autoCleanUp(SharedCallCounterMapTest.java:57)
```

**Comment by ejona86** [comment]
Thank you!

**Pull request merged**
